File: Easy/14_Longest_Common_Prefix/Java_Solutions/14.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        String longst = strs[0];

        for(int i = 0; i < strs.length; i++){
            int j = 0;
            while(j < longst.length() && j < strs[i].length() && longst.charAt(j) == strs[i].charAt(j)){ j++; }

            longst = longst.substring(0,j);

            if(longst.isEmpty()){
                return "";
            }
        }
        return longst;
    }
}

File: Easy/14_Longest_Common_Prefix/Python_Solutions/14.py
from typing import List

class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        longst = strs[0]
        for item in strs[1:]:
            i = 0
            while i < len(longst) and i < len(item) and longst[i] == item[i]:
                i +=1
            longst = longst[:i]
            if longst == "":
                return ""
        return longst